source/mir/ndslice/slice.d

```diff
@@ -513,10 +514,19 @@ createSlice(T,
     size_t N)(auto ref in size_t[N] lengths)
 {
     immutable len = lengthsProduct(lengths);
-    assert(len, "Cannot create empty slice");
     return new T[len].sliced!replaceArrayWithPointer(lengths);
 }
 
+/// ditto
+auto createSlice(
+    Flag!`replaceArrayWithPointer` replaceArrayWithPointer = Yes.replaceArrayWithPointer,
+    size_t N, Range)(auto ref Slice!(N, Range) slice)
+{
+    import std.array: array;
+    import mir.ndslice.selection: byElement;
+    return slice.byElement.array.sliced!replaceArrayWithPointer(slice.shape);
```

Comment:
I meant uses the gc allocation - as said I saw that makeSlice is changed too ;-)